File: android/src/main/java/org/ligi/passandroid/model/comparator/DirectionAwarePassByTimeComparator.java
package org.ligi.passandroid.model.comparator;

import org.ligi.passandroid.model.pass.Pass;

public class DirectionAwarePassByTimeComparator extends PassByTimeComparator {

    public final static int DIRECTION_DESC = -1;
    public final static int DIRECTION_ASC = 1;

    private final int direction;

    public DirectionAwarePassByTimeComparator(int direction) {
        this.direction = direction;
    }

    @Override
    public int compare(Pass lhs, Pass rhs) {
        return super.compare(lhs, rhs) * direction;
    }
}

File: android/src/main/java/org/ligi/passandroid/model/comparator/DirectionAwarePassByTimeComparator.kt
package org.ligi.passandroid.model.comparator

import org.ligi.passandroid.model.pass.Pass

class DirectionAwarePassByTimeComparator(private val direction: Int) : PassByTimeComparator() {

    override fun compare(lhs: Pass, rhs: Pass): Int {
        return super.compare(lhs, rhs) * direction
    }

    companion object {
        val DIRECTION_DESC = -1
        val DIRECTION_ASC = 1
    }
}
